packages/react-accordion/src/components/Accordion/Accordion.tsx

```diff
@@ -7,7 +7,7 @@ import type { AccordionProps } from './Accordion.types';
 /**
  * Define a styled Accordion, using the `useAccordion` and `useAccordionStyles` hooks.
  */
-export const Accordion: React.FunctionComponent<AccordionProps> = React.forwardRef((props, ref) => {
+export const Accordion = React.forwardRef<HTMLDivElement, AccordionProps>((props, ref) => {
```

Comment:
Just doing FunctionComponent the way it was before (and leaving the types off forwardRef) should get basically the same functionality typing-wise and IMO results in a cleaner/more self-documenting API surface. The user doesn't care that something is a ForwardRefExoticComponent, they just want to know what props are available.

For reference, this is FunctionComponent:
```ts
    interface FunctionComponent<P = {}> {
        (props: PropsWithChildren<P>, context?: any): ReactElement<any, any> | null;
        propTypes?: WeakValidationMap<P>;
        contextTypes?: ValidationMap<any>;
        defaultProps?: Partial<P>;
        displayName?: string;
    }
```

This is ForwardRefExoticComponent:
```ts
    interface ForwardRefExoticComponent<P> extends NamedExoticComponent<P> {
        defaultProps?: Partial<P>;
        propTypes?: WeakValidationMap<P>;
    }
    interface NamedExoticComponent<P = {}> extends ExoticComponent<P> {
        displayName?: string;
    }
    interface ExoticComponent<P = {}> {
        /**
         * **NOTE**: Exotic components are not callable.
         */
        (props: P): (ReactElement|null);
        readonly $$typeof: symbol;
    }
```

Reply:
After looking into this some more, I think the best option might be to wrap `React.forwardRef` with a function that gets the types correct. The main problem this PR was trying to solve is that `React.forwardRef` uses `React.PropsWithoutRef`, which breaks the discriminated union.

Instead, we could write our own `forwardRef` wrapper that:
* Accepts a Props type with a `ref` prop already included (no need for `PropsWithoutRef`)
* Returns a cleaner `React.FunctionComponent<Props>` type

It'd be a thin wrapper that just changes forwardRef type. Something like:
```ts
export function forwardRef<Props extends React.RefAttributes<unknown>>(
  component: React.ForwardRefRenderFunction<Props extends React.RefAttributes<infer R> ? R : never, Props>,
) {
  return (React.forwardRef(component) as unknown) as React.FunctionComponent<Props>;
}
```

----

I think I'm going to merge this PR as-is for now, since it fixes a problem that's actively blocking other work. I'll submit a follow-up PR to wrap `forwardRef`, since I expect that might take longer to resolve (wrapping functions is undesirable, and there might be pushback about that...)

Reply:
Follow-up PR here: #19812